File: src/main/java/tn/esprit/backend/services/interfaces/AiCoachService.java
package tn.esprit.backend.services.interfaces;

import tn.esprit.backend.dto.AiCoachPreviewRequest;
import tn.esprit.backend.dto.AiCoachPreviewResponse;
import tn.esprit.backend.entities.Application;

public interface AiCoachService {
    AiCoachPreviewResponse preview(AiCoachPreviewRequest request);
    Application applyImprovedText(Long applicationId, Long applicantId, String improvedText);
}

File: frontend/scholar-1.0.0/education-ui/src/app/core/services/ai-coach.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import {
  AiCoachApplyRequest,
  AiCoachPreviewRequest,
  AiCoachPreviewResponse
} from '../models/ai-coach.model';
import { Application } from '../models/application.model';

@Injectable({ providedIn: 'root' })
export class AiCoachService {
  private base = '/skillhub/api/ai/coach';

  constructor(private http: HttpClient) {}

  preview(request: AiCoachPreviewRequest): Observable<AiCoachPreviewResponse> {
    return this.http.post<AiCoachPreviewResponse>(`${this.base}/preview`, request);
  }

  apply(applicationId: number, applicantId: number, request: AiCoachApplyRequest): Observable<Application> {
    return this.http.post<Application>(`${this.base}/apply/${applicationId}/${applicantId}`, request);
  }
}
